src/components/DropdownV2/DropdownV2.js

```diff
@@ -41,6 +41,11 @@ export default class DropdownV2 extends React.Component {
      */
     itemToString: PropTypes.func,
 
+    /**
+     * Optional function to render items as custom components instead of strings.
+     */
+    itemToElement: PropTypes.func,
```

Comment:
As a convention, should we have functions called for rendering be called `renderX`? For this one, could be `renderItem`, potentially?